src/server/utils/bucket_diff.js

```diff
@@ -68,47 +68,67 @@ class BucketDiff {
+            delete_markers: first_bucket_delete_markers
         } = await this.get_objects(this.first_bucket, prefix, max_keys, current_first_bucket_cont_token));
 
         if (first_bucket_cont_token) diff.first_bucket_cont_token = first_bucket_cont_token;
-        if (Object.keys(first_bucket_contents_left).length === 0) return diff;
+        if (Object.keys(first_bucket_contents_left).length === 0 &&
+            Object.keys(first_bucket_delete_markers).length === 0) return diff;
 
         let second_bucket_contents_left;
         let second_bucket_cont_token;
+        let second_bucket_delete_markers;
         let new_second_bucket_cont_token = current_second_bucket_cont_token;
         let keep_listing_second_bucket = true;
 
         while (keep_listing_second_bucket) {
             ({
                 bucket_contents_left: second_bucket_contents_left,
-                bucket_cont_token: new_second_bucket_cont_token
+                bucket_cont_token: new_second_bucket_cont_token,
+                delete_markers: second_bucket_delete_markers
             } = await this.get_objects(this.second_bucket, prefix, max_keys, new_second_bucket_cont_token));
 
             const keys_diff_response = await this.get_keys_diff(
-                first_bucket_contents_left, second_bucket_contents_left, new_second_bucket_cont_token);
+                first_bucket_contents_left, second_bucket_contents_left, new_second_bucket_cont_token,
+                first_bucket_delete_markers, second_bucket_delete_markers);
 
             first_bucket_contents_left = keys_diff_response.keys_contents_left;
+            first_bucket_delete_markers = keys_diff_response.delete_markers_left;
             keep_listing_second_bucket = keys_diff_response.keep_listing_second_bucket;
             diff.keys_diff_map = { ...diff.keys_diff_map, ...keys_diff_response.keys_diff_map };
+            diff.keys_del_diff_map = {...diff.keys_del_diff_map, ...keys_diff_response.keys_del_diff_map};
 
             const first_bucket_key_array = Object.keys(first_bucket_contents_left);
+            const first_bucket_del_array = Object.keys(first_bucket_delete_markers);
             const second_bucket_key_array = Object.keys(second_bucket_contents_left);
 
-            if (first_bucket_key_array.length !== 0 && second_bucket_key_array.length !== 0) {
+            if (first_bucket_key_array.length !== 0 && second_bucket_key_array.length !== 0 ||
+                first_bucket_del_array.length !== 0) {
                 const first_bucket_key_in_last_pos = first_bucket_key_array[first_bucket_key_array.length - 1];
                 const second_bucket_key_in_last_pos = second_bucket_key_array[second_bucket_key_array.length - 1];
-                if (first_bucket_key_in_last_pos >= second_bucket_key_in_last_pos) {
+                if (first_bucket_key_in_last_pos >= second_bucket_key_in_last_pos ||
+                    first_bucket_del_array.length !== 0) {
```

Comment:
Do we need to also add second_bucket_delete_markers.length !== 0 for the case that we only have delete markers?